### DISCUSSION TITLE
Connection refused in development mode on Linux

### DISCUSSION THREAD

**Discussion by bluepostit** [vercel/next.js#49292] [Help]
### Summary

Hi there.

I have created a new Next app with `create-next-app@latest` on Manjaro, which seems to have built successfully.
When I run `npm run dev`, I see no errors or warnings, only output that things are running as expected.

However, when I open `http://localhost:3000` in my browser, I see an error message in the browser. The server logs show:

```txt
error - Error: connect ECONNREFUSED 127.0.0.1:39477
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1283:16) {
  errno: -111,
  code: 'ECONNREFUSED',
  syscall: 'connect',
  address: '127.0.0.1',
  port: 39477
}
```

The app will not load or render the page.

Interestingly, the port shown (here 39477) is different every time I restart the server.

It seems to me that Next is looking for some service which should be listening on some other port, but can't find it. I have no idea what this service might be. I've spent some time searching for similar issues other people have had, but with no success. Can anyone help, please?

It's worth noting that when running the same created app on a Mac, the app runs just fine.
![Screenshot from 2023-05-05 13-31-13](https://user-images.githubusercontent.com/32864499/236446911-9eb4e975-7866-44d6-b6c6-d6f965a4105f.png)

### Edit: clarification

The app is serving on port 3000 as expected. The **client** connects to the server on this port successfully.

However, the **server** is attempting to connect to an **additional port** on the local host. As I wrote above, the number of this port changes each time I restart `npm run dev`

**Comment by icyJoseph** [comment]
When I run the app there's nothing running on that port. 

Have you made modifications to your hosts file? feels like a network issue somehow.

**Reply by bluepostit** [reply]
Thanks for writing. No, I haven't made any changes to the hosts file. 

To clarify, the app is serving on port 3000 as expected. The **client** connects to the server on this port successfully.

However, the **server** is attempting to connect to an additional port on the local host. As I wrote, the number of this port changes each time I restart `npm run dev`

**Reply by icyJoseph** [reply]
Aha Sorry I skimmed through on first read. Could you try to disable telemetry?https://nextjs.org/telemetry#how-do-i-opt-out

**Reply by bluepostit** [reply]
No problem!

Ok, I tried that but still get the same error.

**Reply by icyJoseph** [reply]
mmm then I am a bit out of ideas... I am not 100% sure about what kind of Next.js process,  would cause this issue - maybe it is trying to connect to a language server? Or a socket? Really odd.

**Reply by bluepostit** [reply]
Thanks all the same. Yeah, I have no idea. And it's strange that this is not an issue when I try it on a Mac

**Reply by DenisJulio** [reply]
I have the same issue as described by @bluepostit, also on a linux. It started showing this since `next` 13.3.0 until 13.2.x everything was working fine. Maybe this could be indeed a bug right?

**Reply by bluepostit** [reply]
I opened #49526

**Reply by boncz92** [reply]
Same issue here, someone on SO said to remove the dependency on fonts/google in the initial layout.tsx but that wouldnt make much sense to be the fix. 

Has anyone made any headway with this issue?

**Comment by dav1app** [comment]
Same here. 

Reverting it back to `Node v16` fixed the issue for me. 

This seams to be a race condition in Webpack regarding module load / server rendering.
